src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,118 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.connection.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES(?,?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            int affectedRows = preparedStatement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Can't insert book to the table " + book);
+            }
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create book" + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        Optional<Book> result = Optional.empty();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                result = Optional.of(getBookFromResultSet(resultSet, id));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book with id " + id, e);
+        }
+        return result;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Long id = resultSet.getLong("id");

```

Comment:
See [common mistakes file ](https://github.com/mate-academy/jv-jdbc-intro/blob/main/checklist.md)
good practice: 
```suggestion
                Long id = resultSet.getObject("id", Long.class);
```